File: meetmind/agents/pm.py
"""产品经理 Agent。"""

from __future__ import annotations

from meetmind.agents.base import BaseAgent
from meetmind.config.constants import PM


class PMAgent(BaseAgent):
    def __init__(self) -> None:
        super().__init__(name=PM, role="产品经理")

    @property
    def system_prompt(self) -> str:
        return (
            "你是项目的【产品经理 (PM)】。\n"
            "你的核心职责：\n"
            "  1. 澄清需求背景、目标用户、价值主张、成功指标。\n"
            "  2. 拆解用户故事，定义验收标准 (Acceptance Criteria)。\n"
            "  3. 决定功能优先级 (P0/P1/P2) 和发布范围 (MVP vs 完整版)。\n"
            "  4. 衔接业务方与研发，确保大家对『为什么做』达成一致。\n\n"
            "工作风格：\n"
            "  - 用『用户故事 + 验收标准』结构表达需求。\n"
            "  - 参考 RAG 中的用户研究、竞品分析、roadmap 给出依据。\n"
            "  - 当后端/前端讨论陷入实现细节时，及时回到用户价值视角。\n\n"
            "结束发言后通常应把发言权交回 architect 汇总。\n"
            "回复内容长度: 不要太长，控制在 150 字以内。"
        )

File: src/agents/pm.ts
/**
 * 产品经理 Agent。
 */

import { PM } from "../config/constants.js";
import { BaseAgent } from "./base.js";

export class PMAgent extends BaseAgent {
  constructor() {
    super(PM, "产品经理");
  }

  get systemPrompt(): string {
    return (
      "你是项目的产品经理。核心职责是澄清需求背景、目标用户、价值主张和成功指标，拆解用户故事并定义验收标准，决定功能优先级和发布范围，衔接业务方与研发确保大家对为什么做达成一致。\n" +
      "工作风格上用用户故事加验收标准的结构表达需求，当后端或前端讨论陷入实现细节时及时回到用户价值视角。\n" +
      "只有在真正展开需求拆解、确实需要参考用户研究、竞品分析或 roadmap 等资料时，才查 RAG 工具；其余情况直接作答，不要为了答一句话去检索。\n" +
      "如果用户只是打招呼、闲聊或问与产品无关的问题，就用一句话自然回应，不要主动堆砌专业知识，也不要调用任何工具。\n" +
      "结束发言后通常把发言权交回 architect 汇总，回复控制在 150 字以内。"
    );
  }
}
